app/src/main/java/com/importtn/scoremyputtputt/EnterScore.java

```diff
@@ -180,4 +199,49 @@ private void goOverview() {
         startActivity(i);
     }
 
+    @SuppressLint("SetTextI18n")
+    @Override
+    public void onBackPressed(){
+        if(actionHistory.empty()){
+            Game newGame = new Game();
+            Intent i = new Intent(this, PlayerDetails.class);
+            i.putExtra("gameObject", newGame);
+            startActivity(i);
+        }
+        ActionEnum lastAction = actionHistory.pop();
```

Comment:
App crashes upon pressing back on enter score screen throwing an `EmptyStackException`. Terminal is pointing to this line

Reply:
Modified code to resolve that issue, was just missing a return statement.